dtbase/models/arima.py

```
@@ -0,0 +1,550 @@
+    ) -> dict:
+        """
+        Prepare the training data. Calls the function `prepare_data` from the
+        `dataprocessor` module. Please check that docstring for more details.
+        """
+        return prepare_data(cleaned_data, config=self.config, *args, **kwargs)
+
+    def fit(self, train_data: pd.Series) -> SARIMAXResultsWrapper:
+        """
+        Fit a SARIMAX statsmodels model to a
+        training dataset (time series).
+        The model parameters are specified through the
+        `arima_order`, `seasonal_order` and `trend`
+        settings in config_arima.ini.
+
+        Parameters:
+            train_data: a pandas Series containing the
+                training data on which to fit the model.
+
+        Returns:
+            model_fit: the fitted model, which can now be
+                used for forecasting.
+        """
+        arima_config = self.config["arima"]
+        model = SARIMAX(
+            train_data,
+            order=arima_config.arima_order,
+            seasonal_order=arima_config.seasonal_order,
+            trend=arima_config.trend,
+        )
+        model_fit = model.fit(
+            disp=False
+        )  # fits the model by maximum likelihood via Kalman filter
+        return model_fit
+
+    def get_forecast_timestamp(self, data: pd.Series) -> pd.Timestamp:
+        """
+        Return the end-of-forecast timestamp.
+
+        Parameters:
+            data: pandas Series containing a time series.
+                Must be indexed by timestamp.
+            arima_config: A ConfigArima object containing parameters for the model.
+
+        Returns:
+            forecast_timestamp: end-of-forecast timestamp,
+                calculated by adding the `hours_forecast`
+                parameter of config_arima.ini to the last timestamp
+                of `data`.
+        """
+        arima_config = self.config["arima"]
+        if arima_config.hours_forecast <= 0:
+            logger.error(
```

Same as above, could we just raise an exception with a string argument?